File: leetecode/InvertBinaryTree.py
class Solution:
    def helper(self,root):
        if(root):
            root.left,root.right = root.right,root.left
            self.helper(root.left)
            self.helper(root.right)
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.helper(root)
        return root

File: leetecode/Invertbinarytree.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};
void rec(TreeNode* root)
{
    if(!root)
    {
        return;
    }
    TreeNode* temp = root->left;
    root->left = root->right;
    root->right = root->left;
    rec(root->left);
    rec(root->right);
}
TreeNode* invertTree(TreeNode* root) {
    rec(root);
    return root;
}
int main()
{
    auto left = new TreeNode(2);
    auto right = new TreeNode(7);
    auto root = new TreeNode(4,left,right);
    invertTree(root);
    return 0;
}
